internal/node/node_repositories.bzl

```diff
@@ -567,6 +587,9 @@ def node_repositories(
         minimum_bazel_version = "0.21.0",
     )
 
+    # This "nodejs" repo is just for convinience so one does not have to target @nodejs_<os_name>//...
+    # At macro time we can unfortunately not figure out the host os so that means there will be a
+    # duplicate set of external repositories
```

Comment:
I suppose this means that for an initial clean build the nodejs archive will get downloaded twice? Small price to pay I supposed for toolchain support.

Reply:
As long as the sha is provided (which it is per default) we will only download it once, but yes there is duplicate work with the tar extraction etc and if the `--experimental_repository_cache_hardlinks` flag is not used then it will also take up twice the space on disk. But yeah I also thought it is a worthy tradeoff and I could not think of any other way how we currently could do this.

 If bazel was to support a WORKSPACE alias as requested here https://github.com/bazelbuild/bazel/issues/3219 then I think we could have another repository rule which would generate a `.bzl` file with the correct alias, that the user then would then import as an alternative.

Or maybe actually now that I am thinking about it we could do something similar also without an alias. I will have to play around with it if what I am thinking can actually work out and would improve the situation.

Reply:
Nice. I didn't realize `download_and_extract` uses the repository cache. Cool 👍  That elimates most of the duplication of effort.

Reply:
Nice fix with the alias